kieker-extension-cassandra/src/kieker/extension/cassandra/reader/CassandraDb.java

```diff
@@ -0,0 +1,164 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+package kieker.extension.cassandra.reader;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.exceptions.AuthenticationException;
+import com.datastax.driver.core.exceptions.InvalidQueryException;
+import com.datastax.driver.core.exceptions.NoHostAvailableException;
+import com.datastax.driver.core.exceptions.QueryExecutionException;
+import com.datastax.driver.core.exceptions.QueryValidationException;
+import com.datastax.driver.core.exceptions.UnsupportedFeatureException;
+import com.datastax.driver.core.policies.DowngradingConsistencyRetryPolicy;
+
+import kieker.common.exception.ConfigurationException;
+
+/**
+ *
+ * @author Armin Moebius, Sven Ulrich
+ *
+ */
+public class CassandraDb {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(CassandraDb.class);
+
+	private Cluster cluster;
+	private final List<InetSocketAddress> contactPoints;
+	private final String keyspace;
+	private Session session;
+	private final Map<String, PreparedStatement> preparedStatements = new ConcurrentHashMap<>();
+
+	/**
+	 * Creates a new instance of this class using the given parameter.
+	 *
+	 * @param keyspace
+	 * @param contactPoints
+	 */
+	public CassandraDb(final String keyspace, final List<InetSocketAddress> contactPoints) {
+		this.contactPoints = contactPoints;
+		this.keyspace = keyspace;
+	}
+
+	/**
+	 * Establishes a connection to the database.
+	 *
+	 * @return
+	 */
+	public void connect() {
+		try {
+			this.cluster = Cluster.builder().addContactPointsWithPorts(this.contactPoints)
+					.withRetryPolicy(DowngradingConsistencyRetryPolicy.INSTANCE)
+					.withMaxSchemaAgreementWaitSeconds(60)
+					.build();
+
+			this.session = this.cluster.connect(this.keyspace);
+		} catch (final NoHostAvailableException | AuthenticationException | InvalidQueryException | IllegalStateException exc) {
+			LOGGER.error("Opening Connection to Database failed. {}", exc.getLocalizedMessage());
+		}
+	}
+
+	/**
+	 * Closes all open sessions and connections to the database.
+	 */
+	public void disconnect() {
+		this.session.close();
+		this.cluster.close();
+	}
+
+	/**
+	 * Select data form the Cassandra cluster.
+	 *
+	 * @param fields
+	 * @param table
+	 * @param whereClause
+	 * @return returns a result set
+	 */
+	public ResultSet select(final List<String> fields, final String table, final String whereClause) {
+		String statement = "select ";
```

Comment:
Would be nicer to use Stringbuilder here (or something even more sophisticated for building query strings)

Reply:
Updated this to StringBuilder.